enhancements/builds/disable-jenkins-pipeline.md

```diff
@@ -0,0 +1,299 @@
+The downside of this approach is that the cluster cannot be upgraded to the next minor release.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. How many clusters use Jenkins pipeline builds and also enable a supported feature gate?
+   None today, but that may change if other build related features require a feature gate (example - tech preview items).
+
+### Test Plan
+
+CI already has an existing test suite for the JenkinsPipeline build strategy.
+Setup for this suite will need to enable the `DeprecatedJenkinsPipeline` feature set before any tests run.
+A separate test will need to run against a standard OpenShift cluster that verifies JenkinsPipeline builds fail immediately.
+
+Tests for the `oc` command line will need to be updated as follows:
+
+* JenkinsPipeline builds started via `oc start-build` should issue a warning.
+* JenkinsPipeline BuildConfigs created via `oc new-build` should issue a warning.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Tech Preview -> GA
+
+Not required - this continues the deprecation of a GA feature.
+
+#### Removing a deprecated feature
+
+The Jenkins pipeline strategy has been officially deprecated since OCP 4.1.
+This proposal does not permanently disable the strategy - rather it discourages its use by requiring admins to enable a feature set.
+This will be documented in all places where Jenkins is referenced.
+
+Eventually we would like to make this feature set a `NoUpgrade` feature set.
+Doing so would keep the Jenkins pipeline strategy enabled, but prevent the cluster from being upgraded to the next OCP minor version.
+Hypothetically this is possible:
+
+* In OCP 4.N the feature set is renamed from `DeprecatedJenkinsPipeline` to `DeprecatedJenkinsPipelineNoUpgrade`
+* openshift-controller-manager-operator for OCP 4.N automatically renames the configured feature set from `DeprecatedJenkinsPipeline` to `DeprecatedJenkinsPipelineNoUpgrade`.
+* Once the upgrade to 4.N is complete, the `*NoUpgrade` feature set automatically marks the cluster as not upgradeable.
+* In OCP 4.(N+1) the Jenkins pipline build strategy can be removed in its entirety.
+
+Complete removal would depend on the number and nature of clusters which enable this feature set.
+This would require sign off from product management and relevant business units.
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, previously enabled feature sets will remain enabled.
+The bootstrap logic will ensure that clusters which rely on Jenkins pipeline builds continue to function, unless that cluster has enabled a separate feature set.
+
+On downgrade, clusters with the DeprecatedJenkinsPipeline feature set will continue to function, but will not be upgradeable because the feature set is not known.
+Admins will need to manually reset clusters to use the default feature set.
```

Comment:
does that mean they'll also need to reset the bootstrap check before they upgrade, so that the bootstrap logic can run again and re-assert the featureset if they need it, on upgrade?